In [4]:
import pandas as pd

df = pd.read_csv(r"C:\Users\Netizens\OneDrive\Desktop\Employee_Salary_Analysis\Dataset\WA_Fn-UseC_-HR-Employee-Attrition.csv")

df.head()

,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,...,1,80,0,8,0,1,6,4,0,5
1,49,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,...,4,80,1,10,3,3,10,7,1,7
2,37,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,...,2,80,0,7,3,3,0,0,0,0
3,33,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,...,3,80,0,8,3,3,8,7,3,0
4,27,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,...,4,80,1,6,3,3,2,2,2,2


In [5]:
# Dataset Information
df. info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 35 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Age                       1470 non-null   int64 
 1   Attrition                 1470 non-null   object
 2   BusinessTravel            1470 non-null   object
 3   DailyRate                 1470 non-null   int64 
 4   Department                1470 non-null   object
 5   DistanceFromHome          1470 non-null   int64 
 6   Education                 1470 non-null   int64 
 7   EducationField            1470 non-null   object
 8   EmployeeCount             1470 non-null   int64 
 9   EmployeeNumber            1470 non-null   int64 
 10  EnvironmentSatisfaction   1470 non-null   int64 
 11  Gender                    1470 non-null   object
 12  HourlyRate                1470 non-null   int64 
 13  JobInvolvement            1470 non-null   int64 
 14  JobLevel                

In [6]:
# Dataset Statistics
df. describe()

,Age,DailyRate,DistanceFromHome,Education,EmployeeCount,EmployeeNumber,EnvironmentSatisfaction,HourlyRate,JobInvolvement,JobLevel,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
count,1470.000000,1470.000000,1470.000000,1470.000000,1470.0,1470.000000,1470.000000,1470.000000,1470.000000,1470.000000,...,1470.000000,1470.0,1470.000000,1470.000000,1470.000000,1470.000000,1470.000000,1470.000000,1470.000000,1470.000000
mean,36.923810,802.485714,9.192517,2.912925,1.0,1024.865306,2.721769,65.891156,2.729932,2.063946,...,2.712245,80.0,0.793878,11.279592,2.799320,2.761224,7.008163,4.229252,2.187755,4.123129
std,9.135373,403.509100,8.106864,1.024165,0.0,602.024335,1.093082,20.329428,0.711561,1.106940,...,1.081209,0.0,0.852077,7.780782,1.289271,0.706476,6.126525,3.623137,3.222430,3.568136
min,18.000000,102.000000,1.000000,1.000000,1.0,1.000000,1.000000,30.000000,1.000000,1.000000,...,1.000000,80.0,0.000000,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000
25%,30.000000,465.000000,2.000000,2.000000,1.0,491.250000,2.000000,48.000000,2.000000,1.000000,...,2.000000,80.0,0.000000,6.000000,2.000000,2.000000,3.000000,2.000000,0.000000,2.000000
50%,36.000000,802.000000,7.000000,3.000000,1.0,1020.500000,3.000000,66.000000,3.000000,2.000000,...,3.000000,80.0,1.000000,10.000000,3.000000,3.000000,5.000000,3.000000,1.000000,3.000000
75%,43.000000,1157.000000,14.000000,4.000000,1.0,1555.750000,4.000000,83.750000,3.000000,3.000000,...,4.000000,80.0,1.000000,15.000000,3.000000,3.000000,9.000000,7.000000,3.000000,7.000000
max,60.000000,1499.000000,29.000000,5.000000,1.0,2068.000000,4.000000,100.000000,4.000000,5.000000,...,4.000000,80.0,3.000000,40.000000,6.000000,4.000000,40.000000,18.000000,15.000000,17.000000


In [8]:
# Employee Attrition Count
df["Attrition"].value_counts()

Attrition
No     1233
Yes     237
Name: count, dtype: int64

In [9]:
import numpy as np

# Average Age
print("Average Age:", np.mean(df["Age"]))



Average Age: 36.923809523809524


In [10]:
# Average Monthly Income
print("Average Monthly Income:", np.mean(df["MonthlyIncome"]))

# Maximum Monthly Income
print("Highest Monthly Income:", np.max(df["MonthlyIncome"]))

# Minimum Monthly Income
print("Lowest Monthly Income:", np.min(df["MonthlyIncome"]))

Average Monthly Income: 6502.931292517007
Highest Monthly Income: 19999
Lowest Monthly Income: 1009


In [11]:
# Department-wise Employee Count
df ["Department"] . value_counts()

Department
Research & Development    961
Sales                     446
Human Resources            63
Name: count, dtype: int64

In [15]:
df.groupby("Department")["MonthlyIncome"].mean()

Department
Human Resources           6654.507937
Research & Development    6281.252862
Sales                     6959.172646
Name: MonthlyIncome, dtype: float64

In [18]:
df . groupby(["Department" , "Attrition"]).size()

Department              Attrition
Human Resources         No            51
                        Yes           12
Research & Development  No           828
                        Yes          133
Sales                   No           354
                        Yes           92
dtype: int64

In [19]:
# check Missing Values
df.isnull().sum()

Age                         0
Attrition                   0
BusinessTravel              0
DailyRate                   0
Department                  0
DistanceFromHome            0
Education                   0
EducationField              0
EmployeeCount               0
EmployeeNumber              0
EnvironmentSatisfaction     0
Gender                      0
HourlyRate                  0
JobInvolvement              0
JobLevel                    0
JobRole                     0
JobSatisfaction             0
MaritalStatus               0
MonthlyIncome               0
MonthlyRate                 0
NumCompaniesWorked          0
Over18                      0
OverTime                    0
PercentSalaryHike           0
PerformanceRating           0
RelationshipSatisfaction    0
StandardHours               0
StockOptionLevel            0
TotalWorkingYears           0
TrainingTimesLastYear       0
WorkLifeBalance             0
YearsAtCompany              0
YearsInCurrentRole          0
YearsSince

In [20]:
#check Duplicate Records
df.duplicated().sum()

0

In [21]:
#Average Age
print("Average Age:",
      np.mean(df["Age"]))

Average Age: 36.923809523809524


In [22]:
# Gender Count
df["Gender"].value_counts()

Gender
Male      882
Female    588
Name: count, dtype: int64

In [23]:
# job Role count
df["JobRole"].value_counts()

JobRole
Sales Executive              326
Research Scientist           292
Laboratory Technician        259
Manufacturing Director       145
Healthcare Representative    131
Manager                      102
Sales Representative          83
Research Director             80
Human Resources               52
Name: count, dtype: int64

In [24]:
# Education wise  Employee Count
df["Education"].value_counts()

Education
3    572
4    398
2    282
1    170
5     48
Name: count, dtype: int64

In [25]:
#Maretial Status
df["MaritalStatus"].value_counts()

MaritalStatus
Married     673
Single      470
Divorced    327
Name: count, dtype: int64

In [26]:
#Overtime Count
df["OverTime"].value_counts()

OverTime
No     1054
Yes     416
Name: count, dtype: int64

In [27]:
#Average Total Working Years 
print (np.mean(df["TotalWorkingYears"]))

11.279591836734694


In [29]:
#job Satisfaction Average
print(np.mean(df["JobSatisfaction"]))

2.7285714285714286


In [30]:
# avrage monthly income
avg_income = np.mean(df["MonthlyIncome"])

if avg_income > 5000:
    print("Average salary is above 5000")
else:
    print("Average salary is below 5000")

Average salary is above 5000


In [31]:
def average_salary():
    avg = np.mean(df["MonthlyIncome"])
    return avg

print("Average Salary:", average_salary())

Average Salary: 6502.931292517007


In [32]:
# loop concept
departments = df["Department"].unique()

for dept in departments:
    print(dept)

Sales
Research & Development
Human Resources


In [35]:
try:
    df = pd.read_csv(r"C:\Users\Netizens\OneDrive\Desktop\Employee_Salary_Analysis\Dataset\WA_Fn-UseC_-HR-Employee-Attrition.csv")
    print("Dataset Loaded Successfully")
except FileNotFoundError:
    print("File Not Found")

Dataset Loaded Successfully


# Business Insights

- Total Employees: 1470
- Employees Left Company: 237
- Attrition Rate: Around 16%
- Research & Development has the highest number of employees.
- Average Monthly Income: Around 6503.
- No missing values found in the dataset.
- Department-wise salary analysis completed.